### ISSUE TITLE
Map coords and images still NOT in sync 

### ISSUE THREAD

**Issue by reteP-riS** [JustCarmen/webtrees-fancy-imagebar#80]
https://github.com/JustCarmen/webtrees-fancy-imagebar/issues/51 is still not resolved. See https://www.webtrees.net/index.php/en/forum/help-for-2-0/36087-jc-fancy-imagebar-map-coords-and-images-not-in-sync

Root cause: In ./modules_v4/jc-fancy-imagebar/resources/views/script.phtml there is this line

`var height = $('.jc-fancy-imagebar').height(); // current height of the imagebar`

which is supposed to provide the height of the imagebar, but it provides the height of div.jc-fancy-imagebar, which - maybe accidentally - includes the div.jc-fancy-imagebar-divider with margin-top:3px and height:1px which increases the height of the div.jc-fancy-imagebar by 4px.

There are two solutions to this: either reduce the repective height value by 4px manually by replacing above line of code with

`var height = $('.jc-fancy-imagebar').height() - 4; // current height of the imagebar minus 4px`

or move the div.jc-fancy-imagebar-divider out of the div.jc-fancy-imagebar and below it instead. I'd prefer this second solution.

**Comment by reteP-riS** [comment]
May I additionally suggest to increase the height of the jc-fancy-imagebar-divider to 2px so it matches with the webtrees default divider. That would of course require a correction of -5 to the height of the jc-fancy-imagebar.

**Comment by JustCarmen** [comment]
I really don't understand your problem. I have also read the posts on the forum but that doesn't help much. The height of the image bar is set once on the settings page and does not vary. So for determining the coordinates of an image the same height is taken at every page refresh. Only the width of the images varies and the coordinates change accordingly with each page refresh.

In your forum post you mentioned that you noticed the issue on my homepage too. But I don't see anything deviating on my site. Every image in the imagebar leads to the correct individual page.

And why has the divider to be part of the clickable image? I don't see the point of that. No one would try to click on the divider to open the individual page of the image below it.

> May I additionally suggest to increase the height of the jc-fancy-imagebar-divider to 2px so it matches with the webtrees default divider. That would of course require a correction of -5 to the height of the jc-fancy-imagebar.

The height of the divider is a theme dependent setting of this module and is only meant for layout.  You didn't tell me which theme your are using, but besides that I don't see a reason to change this at the moment. But of course you are free to make slight changes like this yourself. You  could use the CSS/JS module to achieve what you want.

**Comment by reteP-riS** [comment]
Thanks for looking into this and sorry for not making myself clear enough. I'll try to explain it differently.

1. First of all I am using webtrees 2.0.16, jc-fancy-imagebar-2.0.13 and the webtrees default theme.
2. Let's forget the height of the jc-fancy-imagebar-divider for a while. That's a different topic and not an issue at all.
3. Let's forget what I observed on your site.
4. The issue occurs when the javascript in `./modules_v4/jc-fancy-imagebar/resources/views/script.phtml` is triggered.
5. The issue seems to be specific to the theme and the size of the browser window. 
6. The calculation of `var height = $('.jc-fancy-imagebar').height(); // current height of the imagebar` does NOT always return the correct value. Im my case it returns 94px although I set the imagebar height to 90px in the control panel.
7. The reason seems to be that - at least in my case - it includes the height of the .jc-fancy-imagebar-divider with its margin-top:3px and its height:1px which make it additional 4px. This is caused by the surrounding `<div>...</div>` in `./modules_v4/jc-fancy-imagebar/resources/views/fancy-imagebar.phtml`.

Below examples show that the image has a height of 90px, the .jc-fancy-imagebar has a total hight of 94px, and that the .jc-fancy-imagebar-divider adds those 4px. It further shows where those 4px come from and why the height of the .jc-fancy-imagebar-divider should be 2px instead of 1px in order to match the webtrees default theme.

It all depends on whether the javascript is triggered and whether the recalculation of the coords for the specific screen and theme is correct or not. I'll be happy to give a demo in a screen sharing session that I can arrange for.

![fancy-imagebar](https://user-images.githubusercontent.com/77232439/133932575-964a4174-94c4-44e3-86c7-6a91beefd4e4.jpg)
